htlcswitch/link.go

```diff
@@ -1065,12 +1082,73 @@ out:
 		case msg := <-l.upstream:
 			l.handleUpstreamMsg(msg)
 
+		// A hodl event is received. This means that we now have a
+		// resolution for a previously accepted htlc.
+		case hodlItem := <-l.hodlQueue.ChanOut():
+			hodlEvent := hodlItem.(invoices.HodlEvent)
+			err := l.processHodlQueue(hodlEvent)
+			if err != nil {
+				l.fail(LinkFailureError{code: ErrInternalError},
+					fmt.Sprintf("process hodl queue: %v",
+						err.Error()),
+				)
+				break out
```

Comment:
It depends on how strict we want to be about errors. As this code is only executed for hodl invoices and hodl invoices are newly introduced, I think we should be strict about it. Otherwise we risk both not noticing the error when it happens as well as having to deal with consequential damage from for example not being able to update the commitment tx inside `processHodlQueue`.